doc/develop/coding-guidelines.rst

```diff
@@ -0,0 +1,213 @@
+******************************************
+Coding Guidelines
+******************************************
+
+This will help you to write code with the correct format. In general, we follow the
+guidelines stated in `PEP 8 <https://peps.python.org/pep-0008/>`_. Here we will list the most common practices used by a software
+developer for odemis.
+
+Naming Conventions
+==================
+
+* Class names should be in CamelCase in which multiple words are connected by capitalizing the first letter of the word. For e.g. MyClass().
+
+* Function name should be in snake_case in which the multiple words in small capitalization are joined with underscore where the first word is verb. For e.g. get_image()
+
+* Import modules should not be renamed. Use the original import name. For e.g. import numpy (and not import numpy as np)
+
+ * Exception: import matplotlib.pyplot as plt.
+
+* VAs in odemis should be in mixedCase which is different from PEP 8.
+
+
+Exception : If dependent on outer dependencies which follow different
+conventions. For instance, code inheriting from wxPython classes must follow
+the wxPython style, which follows the wxwidget style from
+typical C++ conventions.
+
+
+For e.g. 1.1 the VAs in odemis have the following structure
+
+.. code-block:: python
+
+ self.pixelSize = model.VigilantAttribute(pxs, unit="m", readonly=True)
+ self.horizontalFoV = model.FloatContinuous(hfv, range=[10e-9, 10e-3],
+                                               unit="m")
+ self.magnification = model.VigilantAttribute(self._hfw_nomag / hfv,
+                                                 unit="", readonly=True)
+
+* Naming of test files follow the below ways to provide consistency for files in test folders (no _init_.py file), which are included in each sub-directories of src folder.
+
+ * A unittest convention is used to name the test class for a module as TestModuleName(unittest.Testcase).
```

Comment:
@tepals, @pieleric should we also have a unit testcase naming convention? [example UnitOfWork_StateUnderTest_ExpectedBehavior](https://osherove.com/blog/2005/4/3/naming-standards-for-unit-tests.html)

Reply:
Why not, it sounds a pretty good idea... but for now we have no convention, so it'll not be very much respected for all the current test cases.